dev/io.openliberty.mcp.internal/src/io/openliberty/mcp/internal/McpTransport.java

```
@@ -269,4 +271,27 @@ public String getRequestIpAddress() {
     public McpProtocolVersion getProtocolVersion() {
         return version;
     }
+
+    public <T> CompletionStage<Void> sendResultAsync(CompletionStage<T> stage) {
+        AsyncContext asyncContex = req.startAsync();
+
+        return stage.handle((result, throwable) -> {
+            try {
+                if (throwable == null) {
+                    sendResponse(result);
+                } else {
+                    sendError((Exception) throwable);
+                }
```

This cast isn't safe. Though we will catch the resulting exception below, the original exception will be lost.

It would be better to change `sendError` to accept `Throwable`.

---

done, thanks